File: Quicksort Algorithm/QuickSort.java
class QuickSort {
    // Partition Method
    int partition(int arr[], int lowIndex, int highIndex) {
        int pivot = arr[highIndex];
        int i = (lowIndex - 1);
        for (int j = lowIndex; j < highIndex; j++) {
            if (arr[j] <= pivot) {
                i++;
                int temp = arr[i];
                arr[i] = arr[j];
                arr[j] = temp;
            }
        }
        int temp = arr[i + 1];
        arr[i + 1] = arr[highIndex];
        arr[highIndex] = temp;
        return i + 1;
    }

    // Sorting Method
    void sort(int arr[], int lowIndex, int highIndex) {
        if (lowIndex < highIndex) {
            int pi = partition(arr, lowIndex, highIndex);
            sort(arr, lowIndex, pi - 1);
            sort(arr, pi + 1, highIndex);
        }
    }

    // Method to print array
    static void printArray(int arr[]) {
        int n = arr.length;
        for (int i = 0; i < n; ++i)
            System.out.print(arr[i] + " ");
        System.out.println();
    }

    // Main Method
    public static void main(String args[]) {
        int arr[] = { 101, 37, 68, 29, 11, 5 };
        int n = arr.length;
        QuickSort ob = new QuickSort();
        ob.sort(arr, 0, n - 1);
        System.out.println("sorted array");
        printArray(arr);
    }
}

File: Quicksort Algorithm/QuickSort.js
function partition(arr, lowIndex, highIndex) {
    const pivot = arr[highIndex];
    let i = lowIndex - 1;
    for (let j = lowIndex; j < highIndex; j++) {
        if (arr[j] <= pivot) {
            i++;
            [arr[i], arr[j]] = [arr[j], arr[i]]; // Swap elements at i and j
        }
    }
    [arr[i + 1], arr[highIndex]] = [arr[highIndex], arr[i + 1]]; // Swap pivot with element at i+1
    return i + 1;
}

function quickSort(arr, lowIndex, highIndex) {
    if (lowIndex < highIndex) {
        const pi = partition(arr, lowIndex, highIndex);
        quickSort(arr, lowIndex, pi - 1);
        quickSort(arr, pi + 1, highIndex);
    }
}

function printArray(arr) {
    const n = arr.length;
    for (let i = 0; i < n; i++) {
        process.stdout.write(arr[i] + " ");
    }
    console.log();
}

function main() {
    const arr = [101, 37, 68, 29, 11, 5];
    const n = arr.length;
    quickSort(arr, 0, n - 1);
    console.log("Sorted array:");
    printArray(arr);
}

main();